### ISSUE TITLE
Streaming leads to full page refresh starting from 13.2.5-canary.4

### ISSUE THREAD

**Issue by lsagetlethias** [vercel/next.js#49790]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:58 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6020
    Binaries:
      Node: 16.20.0
      npm: 8.19.3
      Yarn: 1.22.19
      pnpm: 7.24.3
    Relevant packages:
      next: 13.2.5-canary.4
      eslint-config-next: 13.2.5-canary.4
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 4.9.5
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Static HTML Export (output: "export")

### Link to the code that reproduces this issue

https://github.com/lsagetlethias/nextjs-13.2.5-canary.4-stream-bug

### To Reproduce

```sh
# when it does work
yarn install
yarn dev

# when it doesn't work
# change next version to 13.2.5-canary.9
rm -rf .next
yarn install
yarn dev
```

### Describe the Bug

I built a "infinite" list purely with server side rendering for search results. Everything was great until I upgraded to 13.4.
The goal was to have a list of fetched items with a "Next results" button that add more items to the list.

By dichotomy, I found that the bug was introduced in version 13.2.5-canary.4 (maybe because of "output export" feature, only guessing here).

When clicking on "Next results" the page is fully reloading instead of appending new divs at the bottom.

I reproduced it only once with 13.2.5-canary.4 in a dedicated repro (dunno why only "once". Cache?).
But I found the very same behaviour in 13.2.5-canary.9 which is not happening in 13.2.5-canary.8.
![nextbug13 2 5-canary 9](https://github.com/vercel/next.js/assets/5783789/1cce5289-1742-4fb5-8880-719d28cf575f)

### Expected Behavior

The page should not be fully reloaded

![nextbug13 2 5-canary 8](https://github.com/vercel/next.js/assets/5783789/2369f7f0-898d-47a2-b6b5-fa314fc5be09)

The weird part is that the latest canary version works on the repro but not on my project (the only difference is data fetching happen from DB instead of fetch) (The search page of my project if revelant: https://github.com/SocialGouv/egapro/blob/master/packages/app/src/app/(consultation)/index-egapro/recherche/page.tsx)
![nextbug13 4-canary 0](https://github.com/vercel/next.js/assets/5783789/ca656d8f-0676-4eff-859d-90eab8fef083)

### Which browser are you using? (if relevant)

Chrome 113.0.5672.92 (Official Build) (arm64) 

### How are you deploying your application? (if relevant)

next start self hosted docker. Bug happen also in "next dev"

**Comment by lsagetlethias** [comment]
up

**Comment by lsagetlethias** [comment]
Fixed in 13.4.3

**Comment by lsagetlethias** [comment]
Fixed in 13.4.3
